golang/cosmos/app/app.go

```diff
@@ -829,8 +829,14 @@ func NewAgoricApp(
 func unreleasedUpgradeHandler(app *GaiaApp, targetUpgrade string) func(sdk.Context, upgradetypes.Plan, module.VersionMap) (module.VersionMap, error) {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVm module.VersionMap) (module.VersionMap, error) {
 		app.CheckControllerInited(false)
-		// Record the plan to send to SwingSet
-		app.upgradePlan = &plan
+		app.upgradeDetails = &upgradeDetails{
+			// Record the plan to send to SwingSet
+			Plan: plan,
+			// Core proposals that should run during the upgrade block
+			// These will be merged with any coreProposals specified in the
+			// upgradeInfo field of the upgrade plan ran as subsequent steps
+			CoreProposals: vm.CoreProposalsFromSteps(),
```

Comment:
Can you supply a commented-out example of what real-world use would look like?  I was confused by thinking that somehow `vm.CoreProposalsFromSteps()` was actually supposed to do something useful, rather than just generating an emptyish struct.